### ISSUE TITLE
More collection refactoring: Collection and CollectionFile

### ISSUE THREAD

**Issue by lintool** [lintool/Anserini#128]
@Peilin-Yang I'm looking at the Collection API, and I still don't think we have it quite right. So, we have a `Collection` object representing, say, CW09, right? When we actually perform indexing on a particular WARC file, we do something like:

```
Collection collection = (Collection) collectionClass.newInstance();
collection.prepareInput(inputFile);
... // actual indexing here
collection.finishInput();
```

This means that, for each file, we initialize a separate `Collection` object... but that doesn't make sense, because each file *isn't* a collection.

It seems to be that what we need is a `CollectionFile` interface. A `Collection` gives you a bunch of `CollectionFile` objects, and each one is an iterator over source objects.

Thoughts?

**Comment by Peilin-Yang** [comment]
Yes. I think you're right...
Will refactor once I have time.

**Comment by lintool** [comment]
Thanks. Can you do this CR:
https://github.com/lintool/Anserini/pull/127

I'll merge and you can branch off master, so we don't have merge conflicts...

**Comment by lintool** [comment]
Closing, fixed per #148
